src/services/refactors/moveToNewFile.ts

```diff
@@ -65,20 +65,26 @@ namespace ts.refactor {
         readonly ranges: readonly StatementRange[];
     }
 
-    // Filters imports out of the range of statements to move. Imports will be copied to the new file anyway, and may still be needed in the old file.
     function getStatementsToMove(context: RefactorContext): ToMove | undefined {
         const rangeToMove = getRangeToMove(context);
         if (rangeToMove === undefined) return undefined;
         const all: Statement[] = [];
         const ranges: StatementRange[] = [];
         const { toMove, afterLast } = rangeToMove;
-        getRangesWhere(toMove, s => !isPureImport(s), (start, afterEndIndex) => {
+        getRangesWhere(toMove, isAllowedStatementToMove, (start, afterEndIndex) => {

```

Comment:
On a second thought this is not a good check.. The prologue directives are marked as expression statement with string literal as the text so we need test which with this change will fail:

```ts
////"use strict";
////[|function b() {
////    return this;
////}|]
////"something that should not be copied to new file";
////b();
```
You would want something like to get statements to move to get start index keeping original condition
```
const index = findIndex(statements, statement => !isPrologueDirective(statement));
```

Reply:
I used `takeWhile` instead of filter to copy only prologue directives from the top of a file and ignore, similar to the following `find*`.

https://github.com/microsoft/TypeScript/blob/ea842c411e190b306bad7c7e0aefe61ab67da7c8/src/compiler/factory/utilities.ts#L332-L344

The following test

```ts
////"use strict";
////[|function b() {
////    return this;
////}|]
////"something that should not be copied to new file";
////b();
```

The new file will look like so

```ts
////"use strict";
////export function b() {
////    return this;
////}
```

or 

https://github.com/microsoft/TypeScript/blob/8cc90b723383d4e8f828f6e555a6420d29f772df/tests/cases/fourslash/moveToNewFile_prologueDirectives6.ts#L1-L38

Reply:
👍 bringing back `takeWhile`, or just slicing with `findIndex` sounds like a good plan

Reply:
@andrewbranch Thanks for the review. Oke.